protocol_decoder/src/decoding.rs

```diff
@@ -350,12 +354,26 @@ impl ProcessedBlockTrace {
                 let txn_idx_of_dummy_entry =
                     txn_ir.last().unwrap().txn_number_before.low_u64() as usize + 1;
 
-                // Dummy state will be the state after the final txn.
-                let mut withdrawal_dummy =
-                    create_dummy_gen_input(other_data, extra_data, final_trie_state);
+                // To avoid double hashing the addrs, but I don't know if the extra `Vec`
+                // allocation is worth it.
```

Comment:
I guess this is fine as is for now (we're definitely not in a hot spot anyway here). But there are a few allocations across the lib that could be alleviated I think, so worth keeping track of, I'll open a ticket.